freqtrade/freqai/utils.py

```diff
@@ -191,3 +193,28 @@ def add_feature_trace(fig, fi_df, col):
         fig.update_layout(title_text=f"Best and worst features by importance {pair}")
         label = label.replace('&', '').replace('%', '')  # escape two FreqAI specific characters
         store_plot_file(fig, f"{dk.model_filename}-{label}.html", dk.data_path)
+
+
+def record_params(config: Dict[str, Any], full_path: Path) -> None:
+    """
+    Records run params in the full path for reproducibility
+    """
+    params_record_path = full_path / "run_params.json"
+
+    run_params = {
+        "freqai": config.get('freqai', {}),
+        "timeframe": config.get('timeframe'),
+        "stake_amount": config.get('stake_amount'),
+        "stake_currency": config.get('stake_currency'),
+        "max_open_trades": config.get('max_open_trades'),
+        "pairs": config.get('exchange', {}).get('pair_whitelist')
+    }
+
+    with open(params_record_path, "w") as handle:
+        rapidjson.dump(
+            run_params,
+            handle,
+            indent=4,
+            default=str,
+            number_mode=rapidjson.NM_NATIVE
```

Comment:
i think you will want to use `rapidjson.NM_NATIVE | rapidjson.NM_NAN` as number_mode.

`max_open_trades` will be `inf` if set to -1 in the configuration... 
while it's a non-common config nowadays - it's still a supported mode (static stake, but increasing number of trades until you run out of money).

Reply:
well basically yes.

from log coverage, i can see the function is called ...  (actually by almost every freqai test - which is expected).

a simple test (simply removing the write from the function) however shows that it's actually irrelevant what this function does - as long as it doesn't error ... 

i think to protect ourselfs from future regressions, it might be good to do a very simple assert in one of the tests if "has this file been written" ...